examples/structured_2d_dgsem/elixir_euler_polytropic_coupled.jl

```diff
@@ -0,0 +1,155 @@
+
+coordinates_min1 = (-2.0, -1.0) # minimum coordinates (min(x), min(y))
+coordinates_max1 = (0.0, 1.0) # maximum coordinates (max(x), max(y))
+coordinates_min2 = (0.0, -1.0) # minimum coordinates (min(x), min(y))
+coordinates_max2 = (2.0, 1.0) # maximum coordinates (max(x), max(y))
+
+cells_per_dimension = (32, 32)
+
+mesh1 = StructuredMesh(cells_per_dimension,
+                       coordinates_min1,
+                       coordinates_max1)
+mesh2 = StructuredMesh(cells_per_dimension,
+                       coordinates_min2,
+                       coordinates_max2)
+
+# A semidiscretization collects data structures and functions for the spatial discretization.
+semi1 = SemidiscretizationHyperbolic(mesh1, equations1,
+                                     initial_condition_wave_thermal, solver,
+                                     boundary_conditions = (x_neg = BoundaryConditionCoupled(2,
+                                                                                             (:end,
+                                                                                              :i_forward),
+                                                                                             Float64),
+                                                            x_pos = BoundaryConditionCoupled(2,
+                                                                                             (:begin,
+                                                                                              :i_forward),
+                                                                                             Float64),
+                                                            y_neg = boundary_condition_periodic,
+                                                            y_pos = boundary_condition_periodic))
+
+semi2 = SemidiscretizationHyperbolic(mesh2, equations2,
+                                     initial_condition_wave_polytropic, solver,
+                                     boundary_conditions = (x_neg = BoundaryConditionCoupled(1,
+                                                                                             (:end,
+                                                                                              :i_forward),
+                                                                                             Float64),
+                                                            x_pos = BoundaryConditionCoupled(1,
+                                                                                             (:begin,
+                                                                                              :i_forward),
+                                                                                             Float64),
+                                                            y_neg = boundary_condition_periodic,
+                                                            y_pos = boundary_condition_periodic))
```

Comment:
I know it's annoying, but you could consider adding separate variables `boundary_conditions1 = (x_neg =..., )` and `boundary_conditions2 = (x_neg=...)`
and then pass them into the constructors of `semi1` and `semi2` to avoid this ugliness here...

Reply:
Done.

Reply:
Looks much better, thanks!